src/TextUI/TestRunner.php

```diff
@@ -174,14 +176,31 @@ public function doRun(Test $suite, array $arguments = [], bool $exit = true): Te
             \mt_srand($arguments['randomOrderSeed']);
         }
 
-        if ($arguments['executionOrder'] !== TestSuiteSorter::ORDER_DEFAULT || $arguments['resolveDependencies']) {
-            $sorter = new TestSuiteSorter;
+        if (isset($arguments['cacheResultFile'])) {
+            $cache = new TestResultCache($arguments['cacheResultFile']);
+        }
+
+        if ($arguments['executionOrder'] !== TestSuiteSorter::ORDER_DEFAULT || $arguments['executionOrderDefects'] !== TestSuiteSorter::ORDER_DEFAULT || $arguments['resolveDependencies']) {
+            if (!isset($cache)) {
+                $cache = new TestResultCache;
```

Comment:
why no `cacheResultFile` used here?

Reply:
The cacheResultFile argument is handled right above. There is an edge case that I sometimes used myself where you want to specifiy a cache file to *read* from, but do not overwrite it. The tests for the project itself use this with an example cache in `tests/_files/...`

Another scenario is when using `--stop-on-*` where the result file will only store the defects (and timings) of the tests of the shortened run.

Reply:
> There is an edge case that I sometimes used myself where you want to specifiy a cache file to read from, but do not overwrite it

i would drop that, imo it brings overcomplication and confusion

Reply:
I have changed the tests to not cause the edge case anymore. Will simplify the usage of `cacheResult` and `cacheResultFile` tomorrow. :)